packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -365,20 +366,15 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             "Not enough BatchContexts provided."
         );
 
-        // Take a reference to the queue and its length so we don't have to keep resolving it.
-        // Length isn't going to change during the course of execution, so it's fine to simply
-        // resolve this once at the start. Saves gas.
-        IChainStorageContainer queueRef = queue();
-        uint40 queueLength = _getQueueLength(queueRef);
-
         // Counter for number of sequencer transactions appended so far.
         uint32 numSequencerTransactions = 0;
 
-        // We will sequentially append leaves which are pointers to the queue.
-        // The initial queue index is what is currently in storage.
-        uint40 nextQueueIndex = getNextQueueIndex();
-
         BatchContext memory curContext;
+
+        // cache nextQueueIndex
+        // in order to be safe, nothing should read or write to the nextQueueIndex
+        // storage variable until it is updated with the value from nextQueueIndex_temp.
+        uint40 nextQueueIndex_temp = nextQueueIndex;
```

Comment:
```suggestion
        // Cache the nextQueueIndex.
        // In order to be safe, nothing should read or write to the nextQueueIndex
        // storage variable until it is updated with the value from nextQueueIndex_temp.
        uint40 nextQueueIndex_temp = nextQueueIndex;
```

Reply:
Probably should fix the weird semi-snake, semi-camel casing of this var: `nextQueueIndex_temp`